### DISCUSSION TITLE
v32.0

### DISCUSSION THREAD

**Discussion by westnordost** [streetcomplete/StreetComplete#2899] [Release Discussions]
<p>
  The grant from the <a href="https://prototypefund.de/en/project/streetcomplete/">Prototype Fund</a>
  ended in February, however, there is one last big mega-update I can delight you with before development
  on this app must necessarily wind down:<br/>
  It may not look like a lot, but I basically replaced the whole architecture of the app (#2506, ...)
  to do this. Note that this required dropping most local data, such as unsynced edits, downloaded
  quests and statistics.<br/>Special thanks to @FloEdelmann for helping with the refactor.
  </p>
  <h3>🔙️ Superpowered Undo</h3>
  <ul>
  <li>
    You can now <b>undo any edit</b> that hasn't been uploaded yet:<br/>
    quest answers, split ways, deleted places, hidden quests, created notes and note comments (#1029, #2616)
  </li>
  <li>You can now look at your <b>edit history</b> and undo any recent edit in any order, not just the last one</li>
  <li>It is now shown what exactly is tagged for each edit in the edit history</li>
  <li>You can now undo deletions of places even after you have uploaded it (#2441)</li>
  </ul>
  <h3>⛰️ Improved usability and offline mode</h3>
  <ul>
  <li>
  Immediately unlock new quests based on the one you just solved, <b>even when
  offline</b>. This has been a much requested feature but was simply not possible with the previous
  architecture (#1369, #1510, #1550, #1826, #2254, #2438, #2473, #2474, #2676, #2743...)
  </li>
  <li>
  The above also works when splitting a way. The quests for the split segments will be shown
  immediately
  </li>
  <li>When you scan for quests manually, no cache will be used: you always get the freshest data (#2554)</li>
  <li>When unable to fully upload pictures attached to a note due to a bad connection, the app will try again later (#2102)</li>
  </ul>
  <h3>New Quests</h3>
  <ul>
  <li><i>Is this sport field lit?</i> (#2639, #2737), by @eginhard</li>
  <li><i>What type of barrier is this?</i> (#2739, #2753), by @matkoniecz</li>
  <li><i>What type of stile is this?</i> (#2749, #2766), by @matkoniecz</li>
  <li><i>What type of shop is this?</i> (#2391, #2759), by @matkoniecz</li>
  <li>Italy only: <i>Which type of police station is this?</i> (#2456, #2675), by @naposm</li>
  <li><i>Is this bus stop lit?</i> (#2383, #2846), by @TurnrDev</li>
  </ul>
  <h3>Quest Enhancements</h3>
  <ul>
  <li>Don't ask for building levels for industrial buildings (#2835), by @matkoniecz</li>
  <li>Improve referring to map features by name for some quests (#2840)</li>
  <li>Checking existence: ask for some map features less frequently, by @matkoniecz</li>
  <li>Enable answering that the cycleway is displayed separately on the map (#2525, #2665)</li>
  <li>Some wording improvements (#2728), by @matkoniecz, (#2806, #2814) by @peternewman, (#2685)</li>
  <li>Enhance maxspeed collection for the United Kingdom (#2745, #2748, #2750, #2811), by @arrival-spring</li>
  <li>Periodically ask for the existence of BBQ spots, ticket validators, emergency life rings and emergency phones (#2679), thanks @arrival-spring</li>
  <li>Also ask for the surface of service roads as long as they are not driveways (#2783)</li>
  <li>Don't remove the check_date tag if it is already tagged, instead, update it (#2861, #2883)</li>
  <li>Don't tag foot=use_sidepath on the road if there is a separate sidewalk (#2895)</li>
  </ul>
  <h3>Other Enhancements</h3>
  <ul>
  <li>Added Arabic by Charbel Bechara and others</li>
  <li>Added Romanian by Georgian Iosef and others</li>
  <li>Improvements in the quest selection screen (#2243, #2847, #2828, #2843, #2887), by @FloEdelmann</li>
  <li>Slightly more performant compass</li>
  <li>When showing all notes, really show all except those where the user commented last (#2692)</li>
  </ul>
  <h3>Fixes</h3>
  <ul>
  <li>On tablets, correctly restore the position when reentering the app (#2659, #2311), thanks @matteblair</li>
  <li>Sometimes the link collection wouldn't show (#2801, #2809), by @tapetis</li>
  <li>The keyboard would not appear in the opening hours times dialog (#2799, #2820), by @tapetis</li>
  <li>The flags in the statistics view were sometimes wrongly stretched (#2819, #2831), by @tapetis </li>
  <li>The time picker for the postbox collection times now follows the system settings (#2807)</li>
  <li>Fix crash on trying to display opening hours that range to 24:00 (#2830)</li>
  <li>Remove checking existence of grit bins, as many are removed for summer and not marked as seasonal :( (#2726), by @matkoniecz</li>
  <li>Fix quests that are very close to notes could in rare cases not be opened (#2853)</li>
  <li>Fixes on specifying shop type quest, by @peternewman, @matkoniecz</li>
  <li>...and other fixes and enhancements (#2521, #1821, #2628, #2691, #2715, #2870, #2891...)</li>
  </ul>

<hr /><em>This discussion was created from the release <a href='https://github.com/streetcomplete/StreetComplete/releases/tag/v32.0'>v32.0</a>.</em>

**Comment by Zaijo** [comment]
Amazing piece of work. Thank you.

**Comment by rhhsm** [comment]
Have been using it for a couple of days and my surveying has become far more efficient. Thanks!

**Comment by mnalis** [comment]
Yes, fully offline is even more amazing than I was expecting (and I did not have low expectations!). :cake: 

Relatedly, @westnordost I see we're now at [v33.0-beta1](https://github.com/streetcomplete/StreetComplete/releases/tag/v33.0-beta1), but no new translations have been pushed to POEditor since v32.1? But looking at the changelog, it looks like there **are** new quests which might be needing translations. 

If so, it would be great if new quests could be pushed to POEditor so we translators can get to work

**Reply by westnordost** [reply]
Don't worry, I already uploaded the strings for v33.0 a few weeks ago, just after I released v32.1
